### PR TITLE
Forcing write extended linear address at each beginning of segments

### PR DESCRIPTION
**Pull request by Martoni**
I had a problem with my da14580 chip programmer.

The programmer will not be able to read the HEX file correctly if Extended Linear Address record are not repeated at each beginning of segment.

Here an option to method write_hex_file to add this functionnality.

### CODE REVIEW

**Comment by bialix** [comment]
I need simple unit test for new feature. And I don't quite understand the problem, what do you mean by "repeated at each beginning of segment."

File: intelhex/__init__.py [hunk 1]
```diff
@@ -578,47 +582,6 @@ def write_hex_file(self, f, write_start_addr=True, eolstyle='native'):
             # Python 2
             table = ''.join(chr(i).upper() for i in range_g(256))
 
-        # start address record if any
```
Comment (The-42) [inline intelhex/__init__.py]: Moving this block to the end has nothing to do with this pull request and also breaks the output for many tools.

File: intelhex/__init__.py [hunk 2]
```diff
@@ -693,14 +657,60 @@ def write_hex_file(self, f, write_start_addr=True, eolstyle='native'):
                     # adjust cur_addr/cur_ix
                     cur_ix += chain_len
                     if cur_ix < addr_len:
-                        cur_addr = addresses[cur_ix]
+                        if write_ela and (addresses[cur_ix] - cur_addr > 0x10):
```
Comment (The-42) [inline intelhex/__init__.py]: Do not hardcode ```0x10``` but use ```data_rec_len``` instead. Also check for that parameter in line 593 to get an ELA for shorter hex files.

**Comment by bialix** [comment]
Sorry, I don't quite understand.

> The programmer will not be able to read the HEX file correctly if Extended Linear Address record are not repeated at each beginning of segment.

So each data record should be prepended with extended linear address record? I.e.

```
record 04 addr1
record 00
record 04 addr1
record 00
```

If this is correct that sounds for me really weird and that does not follow what data sheet on IntelHex format said.

**Comment by bialix** [comment]
Sorry for not working on your pull request. I'm looking for a new maintainer for Python IntelHex project. I hope someone will help.

**Comment by The-42** [comment]
Sounds urgent... I'd take it :) Provided your e-mail address from your last commit is valid, you've got a message now.

**Comment by Martoni** [comment]
Sorry for not following this pull request too. In fact I needed this software a long time ago, and have not used it since.

**Comment by bialix** [comment]
I'm still don't really understand the purpose of this patch, and it seems author of patch no more needed it.

**Pull request closed without merging**